### ISSUE TITLE
something wrong about  the training result

### ISSUE THREAD

**Issue by DavidKong96** [ximinng/LLM4SVG#14]
hi, i try to reproduce your result in Qwen-2.5-VL-7B, but i got this result below in 14000 training steps

![Image](https://github.com/user-attachments/assets/1049d547-b573-4cee-ace1-7233857b7420)

could you please let me konw where i did wrong?

I pulled the latest repository, and below is the traning config i use.
```yaml
### model
model_name_or_path: ../weights/Qwen2.5-VL-7B-Instruct
image_max_pixels: 262144
video_max_pixels: 16384
trust_remote_code: true

### method
stage: sft
do_train: true
finetuning_type: lora
lora_rank: 128
lora_target: all
#lora_dropout: 0.1

### dataset
dataset: SVGX_SFT_vision_25k_int_enc
template: qwen2_vl
cutoff_len: 4096
max_samples: 20000000
overwrite_cache: true
preprocessing_num_workers: 128

### output
output_dir: saves/encode-code/qwen2_5vl-7B-lora-sft
logging_steps: 50
save_steps: 2000
plot_loss: true
overwrite_output_dir: true

### train
per_device_train_batch_size: 1
gradient_accumulation_steps: 8
learning_rate: 3.0e-4
num_train_epochs: 5.0
lr_scheduler_type: cosine
warmup_ratio: 0.1
bf16: true
ddp_timeout: 180000000

### new tokens
new_special_tokens_config: data/SVGX-dataset/svg_tokens.yaml
init_special_tokens: semantic_init

### generation
max_length: 4096
max_new_tokens: 4096

### eval
val_size: 0.02
per_device_eval_batch_size: 1
eval_strategy: steps
eval_steps: 2000
```
``
thank you.

**Comment by DavidKong96** [comment]
> Please make sure to include the following setting in the qwen2_vl.yaml file.
> 
> ```
> ### new tokens
> new_special_tokens_config: data/SVGX-dataset/svg_tokens.yaml
> skip_special_tokens: False
> ```

tranks for your reply.
i try to add **new_special_tokens_config: data/SVGX-dataset/svg_tokens.yaml ; skip_special_tokens: False** in qwen2_vl.yaml, and use llamafactory-cli webchat  ./examples/inference/qwen2_vl.yaml to inference, and the [INFO] show below

![Image](https://github.com/user-attachments/assets/be038dbc-5614-4d51-9ab4-3d7caa6fb80e)

but it still not work.

![Image](https://github.com/user-attachments/assets/d38ea390-9414-4cd0-b4e2-4d0654958a5a)

**Comment by DavidKong96** [comment]
> Please make sure to include the following setting in the qwen2_vl.yaml file.
> 
> ```
> ### new tokens
> new_special_tokens_config: data/SVGX-dataset/svg_tokens.yaml
> skip_special_tokens: False
> ```

it works! thank you so much

**Comment by klayc-gzl** [comment]
Why does the svg image I generate always have a small spot in the upper left corner?
like this:

<img width="197" alt="Image" src="https://github.com/user-attachments/assets/1f3659f1-1d65-4cc0-b324-e6433fd68f56" />

**Comment by DavidKong96** [comment]
> Why does the svg image I generate always have a small spot in the upper left corner? like this:
> 
> <img alt="Image" width="197" src="https://private-user-images.githubusercontent.com/80515849/449734952-1f3659f1-1d65-4cc0-b324-e6433fd68f56.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDg5MzcwMzEsIm5iZiI6MTc0ODkzNjczMSwicGF0aCI6Ii84MDUxNTg0OS80NDk3MzQ5NTItMWYzNjU5ZjEtMWQ2NS00Y2MwLWIzMjQtZTY0MzNmZDY4ZjU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjAzVDA3NDUzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE5OGNjM2Y3ZjViN2NkMDViOTY4OTNkZjNmMzhlOWQ4YmI5YzZlZGI3NTRmOTdkZWUwYTU0NTdkODI1MTc2NDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.O-tRpf6wbhrFKEWWdF1veDJOjk9voyof9N3kwv4YvnI">

hi.have you finished the training? i can't reproduce the result  through 128 rank lora train based on qwen-2.5-vl-7b. can you share how you did it ?
thank you !

**Comment by klayc-gzl** [comment]
I also failed to reproduce it, the generated image is very different from the original image. So I want to ask you for help.

**Comment by DavidKong96** [comment]
> I also failed to reproduce it, the generated image is very different from the original image. So I want to ask you for help.

i use llama-factory to reprodece it, and below is my training config and loss curve.

![Image](https://github.com/user-attachments/assets/64362318-4d72-4eb7-86b0-e44646e6ee08)

![Image](https://github.com/user-attachments/assets/2f7056bf-33a6-4046-a6f9-94e792fc7c90)

and the 32000steps inference result like this strange graphics.

![Image](https://github.com/user-attachments/assets/0e00921f-dbfc-45ff-b881-8852cc5b1fbd)

![Image](https://github.com/user-attachments/assets/679b97ac-2609-4cf1-9bee-b991d187365c)

**Comment by klayc-gzl** [comment]
Yes, i use qwen2vl_7b full finetune and have the same loss spike.

![Image](https://github.com/user-attachments/assets/109c0653-ebaf-4276-b716-d201f0cb1082)

**Comment by DavidKong96** [comment]
> Yes, i use qwen2vl_7b full finetune and have the same loss spike.
> 
> ![Image](https://github.com/user-attachments/assets/109c0653-ebaf-4276-b716-d201f0cb1082)

But you got the correct image, just in the wrong position and size?

**Comment by klayc-gzl** [comment]
No, the image is wrong

**Comment by DavidKong96** [comment]
> No, the image is wrong

QAQ...i'm thinking to mix some common data to train.

**Comment by IApple233** [comment]
Hi, I have met the same loss spike when fine tune qwen-7b, do you think this loss curve is correct? because after the spike ,I noticed that the model's performance on eval dataset become worse

**Comment by DavidKong96** [comment]
> Hi, I have met the same loss spike when fine tune qwen-7b, do you think this loss curve is correct? because after the spike ,I noticed that the model's performance on eval dataset become worse

It cannot be said to be correct, but it is indeed a very common phenomenon. the model may already overfitting when loss spike appear

**Comment by yinlikestudy** [comment]
> Why does the svg image I generate always have a small spot in the upper left corner?为什么我生成的 svg 图片左上角总是有一个小斑点？ like this:  像这样：
> 
> <img alt="Image" width="197" src="https://private-user-images.githubusercontent.com/80515849/449734952-1f3659f1-1d65-4cc0-b324-e6433fd68f56.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ1NDI5NzcsIm5iZiI6MTc1NDU0MjY3NywicGF0aCI6Ii84MDUxNTg0OS80NDk3MzQ5NTItMWYzNjU5ZjEtMWQ2NS00Y2MwLWIzMjQtZTY0MzNmZDY4ZjU2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODA3VDA0NTc1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI3ZjZiMzU2ZmNjMzVjM2FmOGM2ODVmZmMwZTkxMzE3YTc2ODM3YTE1MmQ1MGVjZDY2NTc5YzA5Zjc1YzBmZDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Xr562PTQX0ZkuFi9i9cEYPlZyEUHZXM_BZNeJrq2W9Y">

Hello, when reproducing with Qwen2.5VL, I noticed that during img2svg or text2svg inference, the generated images often appear in the top-left corner. Have you encountered this issue?
